### PR TITLE
Gesterhuyse/role management refactor

### PR DESCRIPTION
**Pull request by SuperSouls**
Refactored Role Access Manage Controller and the Role Access Manage Controller Unit Test.

image
image

### CODE REVIEW

File: RR.App/Controllers/HRIS/RoleManageController.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ﻿using HRIS.Models;
 using HRIS.Services.Interfaces;
+using HRIS.Services.Services;
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/RoleManageController.cs:3]:
```suggestion
```

File: RR.App/Controllers/HRIS/RoleManageController.cs [hunk 2]
```diff
@@ -9,6 +11,7 @@ namespace RR.App.Controllers.HRIS;
 [ApiController]
 public class RoleManageController : ControllerBase
 {
+    private readonly AuthorizeIdentity _identity;
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/RoleManageController.cs:14]:
```suggestion
```

File: RR.App/Controllers/HRIS/RoleManageController.cs [hunk 3]
```diff
@@ -9,6 +11,7 @@ namespace RR.App.Controllers.HRIS;
...
     private readonly IRoleAccessLinkService? _roleAccessLinkService;
     private readonly IRoleAccessService? _roleAccessService;
     private readonly IRoleService? _roleService;
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/RoleManageController.cs:17]:
```suggestion
    private readonly IRoleAccessLinkService _roleAccessLinkService;
    private readonly IRoleAccessService _roleAccessService;
    private readonly IRoleService _roleService;
```

I'm not sure why these services are marked as nullable - Should never be nullable. Please change this as well.

File: RR.App/Controllers/HRIS/RoleManageController.cs [hunk 4]
```diff
@@ -60,14 +51,7 @@ public class RoleManageController : ControllerBase
...
-        {
-            return BadRequest("Invalid input");
-        }
-
-        try
-        {
-            var foundRole = await _roleService.CheckRole(role)
+        var foundRole = await _roleService.CheckRole(role)
                 ? await _roleService.GetRole(role)
                 : await _roleService.SaveRole(new RoleDto{ Id = 0, Description = role});
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/RoleManageController.cs:56]:
```suggestion
```

Please remove all unused code

Also please indent your code. `CTRL + K + D`

**Review by EduanO — CHANGES_REQUESTED** [review]

File: RR.App/Controllers/HRIS/RoleManageController.cs [hunk 5]
```diff
@@ -9,9 +10,9 @@ namespace RR.App.Controllers.HRIS;
...
 {
-    private readonly IRoleAccessLinkService? _roleAccessLinkService;
-    private readonly IRoleAccessService? _roleAccessService;
-    private readonly IRoleService? _roleService;
+    private readonly IRoleAccessLinkService _roleAccessLinkService;
+    private readonly IRoleAccessService _roleAccessService;
+    private readonly IRoleService _roleService;
 
     public RoleManageController(
         IRoleAccessLinkService? roleAccessLinkService,
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/RoleManageController.cs:18]:
```suggestion
        IRoleAccessLinkService roleAccessLinkService,
```

This also needs to change. Should not be nullable types

**Review by EduanO — CHANGES_REQUESTED** [review]

**Review by EduanO — APPROVED** [review]

**Review by ReitumetseMoloto — APPROVED** [review]

**Review by MathildaShongwe15 — APPROVED** [review]

**Pull request merged**